From 024a46d300f97f3ee8a5de7f0dc56c044a417354 Mon Sep 17 00:00:00 2001
From: xzhang <xzhang@xzdt>
Date: Thu, 21 Jan 2016 11:32:22 -0800
Subject: [PATCH 0470/1164] CDH-35082: HIVE-12611: Make sure spark.yarn.queue
 is effective and takes the value from
 mapreduce.job.queuename if given [Spark Branch]
 (Rui reviewed by Xuefu)

Change-Id: I59637f63b299ade8fbae1d0ef16689866bb2f1d2
---
 .../hive/ql/exec/spark/HiveSparkClientFactory.java |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
index e02caa7..274dcf0 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
@@ -172,6 +172,15 @@ public static HiveSparkClient createHiveSparkClient(HiveConf hiveconf) throws Ex
     sparkConf.put(
       "spark.kryo.classesToRegister", Joiner.on(",").join(classes));
 
+    // set yarn queue name
+    final String sparkQueueNameKey = "spark.yarn.queue";
+    if (sparkMaster.startsWith("yarn") && hiveConf.get(sparkQueueNameKey) == null) {
+      String queueName = hiveConf.get("mapreduce.job.queuename");
+      if (queueName != null) {
+        sparkConf.put(sparkQueueNameKey, queueName);
+      }
+    }
+
     return sparkConf;
   }
 
-- 
1.7.9.5

